tctl-snap/tctl-plugins/cmd/tctl-authorization/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/canonical/charmed-temporal-image/tctl-plugins/cmd"
+	"github.com/hashicorp/go-plugin"
+	cliplugin "github.com/temporalio/tctl/cli/plugin"
+)
+
+type provider struct{}
+
+func (p provider) GetHeaders(ctx context.Context) (map[string]string, error) {
+	env := os.Getenv("TCTL_ENVIRONMENT")
+	clientID, err := cmd.ClientID()
```

Comment:
i guess you can make this error return only if there is smth set, to avoid having to check if is empty afterwards